src/Symfony/Component/Notifier/Bridge/Mercure/MercureTransportFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mercure;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Mercure\PublisherInterface;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+final class MercureTransportFactory extends AbstractTransportFactory
+{
+    private $publisherLocator;
+
+    /**
+     * @param ContainerInterface $publisherLocator A container that holds {@see PublisherInterface} instances
+     */
+    public function __construct(ContainerInterface $publisherLocator)
+    {
+        parent::__construct();
+
+        $this->publisherLocator = $publisherLocator;
+    }
+
+    /**
+     * @return MercureTransport
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        if ('mercure' !== $dsn->getScheme()) {
+            throw new UnsupportedSchemeException($dsn, 'mercure', $this->getSupportedSchemes());
+        }
+
+        $publisherId = $dsn->getHost();
+        if (!$this->publisherLocator->has($publisherId)) {
+            throw new LogicException(sprintf('"%s" was not found in publishers.', $publisherId));
```

Comment:
Can we say something like:

... not found, did you mean one of: ...... ?

Reply:
I'd like to. But the `Psr\Container\ContainerInterface` only have `get` and `has` methods. Therefore I can't list all available publishers. Maybe there is another way?

Reply:
Oh didn't now that 👍🏻

cc @nicolas-grekas any ideas 💡 or possibilities? Or is it out of scope?

Reply:
Using `ServiceProviderInterface` as type for the `publisherLocator` arg/prop should help (requires `"symfony/service-contracts": "^1.10|^2"`)